### PR TITLE
#17 Escape string parameters instead of converting them to JsString

### PR DESCRIPTION
**Pull request by Gerrrr**

### CODE REVIEW

File: server/src/test/scala/de/frosner/broccoli/templates/TemplateRendererSpec.scala [hunk 1]
```diff
@@ -52,14 +52,14 @@ class TemplateRendererSpec extends Specification with Mockito {
         id = "1",
         template = Template(
           id = "1",
-          template = "{{id}}",
+          template = """"{{id}}"""",
```
Comment (FRosner) [inline server/src/test/scala/de/frosner/broccoli/templates/TemplateRendererSpec.scala:55]: """"ok""""

**Review by FRosner — APPROVED** [review]

File: build.sbt [hunk 2]
```diff
@@ -20,12 +20,12 @@ lazy val server = project
       Dependencies.scalaUri,
       Dependencies.squants,
       Dependencies.scalaguice,
+      Dependencies.commonsIO,
```
Comment (swsnr) [inline build.sbt:23]: The feature you need is from commons lang.  I think that's a separate dependency.

If it is please use commons labg here and keep commons IO as test dependency.

Relying on transitive dependencies is risky.

**Comment by swsnr** [comment]
What's the purpose of this change?

File: build.sbt (hunk 2 above)
Comment (Gerrrr) [reply build.sbt:23]: Thanks for the catch! You are right about commons lang. I'm going to fix it in a separate PR.

**Comment by Gerrrr** [comment]
In #265 all string parameter values are rendered in templates as valid JSON strings - escaped and wrapped by quotes. The issue is that we do not always want to have the quotes directly around the parameter value. Example: [a service name that consist of id and a constant string](https://github.com/FRosner/cluster-broccoli/blob/master/templates/http-server/template.json#L35). If `id` parameter is of string type, the rendered template would look like this - `""service-name"-web-ui-1"`. With the previous implementation it took a lot of effort in the template logic to get rid of the extra pair of quotes around the parameter.

We discussed with @FRosner that it is more convenient and simpler to use if the string type would just make sure that the parameter is an escaped string and it is the responsibility of a template creator to put quotes where they are needed.

**Comment by swsnr** [comment]
@Gerrrr I see; thanks for explaining.

**Pull request merged**